include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -184,37 +184,86 @@ struct __init_processing
 // transform_reduce
 //------------------------------------------------------------------------
 
+template <typename _ExecutionPolicy, typename _Tp, typename _Operation1, typename _Operation2,
+          ::std::size_t __iters_per_work_item = 0>
+struct transform_reduce_known
+{
+  private:
+    // Apply unary and binary operations __n values and return result.
+    template <typename _BinaryOp, typename _UnaryOp, typename... _Acc>
+    static _Tp
+    apply_unary_binary(const ::std::size_t __items_to_process, const ::std::size_t __adjusted_global_id,
```

Comment:
Probably does `constexpr` helps for that function? or we may will rely on argument type deduction for `__items_to_process` (?)

Reply:
I added `constexpr`.

Reply:
I may be wrong, but I don't think `constexpr` helps this function.  

"The constexpr specifier declares that it is possible to evaluate the value of the function or variable at compile time."

As I understand it, we are marking to the compiler that there is some situation where this function could be fully evaluated at compile time (and therefore could be called within a constexpr elsewhere), I don't believe that can ever be true, or even used if it were true.  The data range we are processing would need to be known at compile time.

Does this have some side effect to help to deduce `__items_to_process` at compile time where applicable?

Reply:
Good point about the data range. And I didn't see any change when using `constexpr`.